Zend/Optimizer/zend_func_info.c

```diff
@@ -85,34 +87,23 @@ static uint32_t zend_range_info(const zend_call_info *call_info, const zend_ssa
 	}
 }
 
-static const func_info_t func_infos[] = {
+static const func_info_t old_func_infos[] = {
 	/* zend */
-	F1("zend_version",            MAY_BE_STRING),
-	FN("func_get_args",           MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_ANY),
 	F1("get_class_vars",          MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_STRING | MAY_BE_ARRAY_OF_ANY | MAY_BE_ARRAY_OF_REF),
-	F1("get_class_methods",       MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING),
-	F1("get_included_files",      MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING),
-	FN("set_error_handler",       MAY_BE_NULL | MAY_BE_STRING | MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING | MAY_BE_ARRAY_OF_OBJECT | MAY_BE_OBJECT),
 	F0("restore_error_handler",   MAY_BE_TRUE),
 	F0("restore_exception_handler", MAY_BE_TRUE),
 	F1("get_declared_traits",     MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING),
 	F1("get_declared_classes",    MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING),
 	F1("get_declared_interfaces", MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_OF_STRING),
 	F1("get_defined_functions",   MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_STRING | MAY_BE_ARRAY_OF_ARRAY),
 	F1("get_defined_vars",        MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_STRING | MAY_BE_ARRAY_OF_ANY | MAY_BE_ARRAY_OF_REF),
-	F1("get_resource_type",       MAY_BE_STRING),
-	F1("get_defined_constants",   MAY_BE_ARRAY | MAY_BE_ARRAY_KEY_STRING | MAY_BE_ARRAY_OF_NULL | MAY_BE_ARRAY_OF_FALSE | MAY_BE_ARRAY_OF_TRUE | MAY_BE_ARRAY_OF_LONG | MAY_BE_ARRAY_OF_DOUBLE | MAY_BE_ARRAY_OF_STRING | MAY_BE_ARRAY_OF_RESOURCE | MAY_BE_ARRAY_OF_ARRAY),
```

Comment:
I believe there was a bug in `get_defined_constants()` and `constant()`: `MAY_BE_ARRAY_OF_OBJECT` was forgotten to be added

Reply:
I've fixed that separate in https://github.com/php/php-src/commit/a4c2fb1462c7cf189a7c12570daf8445117fe409.

A question that came up in that context is how we can distinguish "array of ref". For example, get_defined_constants() returns an array with arbitrary values, but it can't contain references...

Reply:
Yes, I was also wondering about it, but I don't have a clear answer yet. Another smaller gotcha is when the return type info is more precise than the type declaration in the stubs (e.g. `restore_error_handler `). I think I can make this case work via PHPDoc: so that `bool` is shown  instead of `true` when generating method synopses

Reply:
I added support for `@return true` as well as improving alias verification. Hopefully the validation of phpdoc types make sense (but at least I got no false positive when running against the root dir).